pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1611,6 +1622,10 @@ public RealtimeSegmentDataManager(SegmentZKMetadata segmentZKMetadata, TableConf
       _segmentLogger
           .info("Starting consumption on realtime consuming segment {} maxRowCount {} maxEndTime {}", llcSegmentName,
               _segmentMaxRowCount, new DateTime(_consumeEndTime, DateTimeZone.UTC));
+      _allowPartialUpsertConsumptionDuringCommit =
+          _realtimeTableDataManager.isPartialUpsertEnabled() ? _tableConfig.getUpsertConfig() != null
```

Comment:
the config name as `allowPartialUpsertConsumptionDuringCommit` or `default.allow.partial.upsert.consumption.during.commit` makes a lot sense and it's false by default.

but this variable name `_allowPartialUpsertConsumptionDuringCommit` here might be a bit less readable. Here we had to set it true for non-paritial-upsert table. I'd call it `_allowConsumptionDuringCommit`, then it's more intuitive to leave it as true for non-paritial-upsert table

```
_allowConsumptionDuringCommit = !_realtimeTableDataManager.isPartialUpsertEnabled()? true : _tableConfig.getUpsertConfig().isAllowPartialUpsertConsumptionDuringCommit() 
// nit: no need to check getUpsertConfig() is not null as isPartialUpsertEnabled() is true already.
```

Reply:
Sounds good. Updated.